src/Microsoft.Azure.WebJobs.Host/Tables/TableEntityWriter.cs

```diff
@@ -172,23 +172,35 @@ private static ITableEntity Copy(ITableEntity item)
                 {
                     if (!e.IsNotFoundTableNotFound())
                     {
-                        throw;
+                        throw new StorageException(e.GetDetailedErrorMessage());
                     }
 
                     exception = e;
                 }
 
                 if (exception != null)
                 {
-                    // Make sure the table exists
-                    await _table.CreateIfNotExistsAsync(cancellationToken);
-
-                    // Commit the batch
-                    await _table.ExecuteBatchAsync(batch, cancellationToken);
+                    await CreateTableAndExecuteBatch(batch, cancellationToken);
                 }
             }
         }
 
+        private async Task CreateTableAndExecuteBatch(IStorageTableBatchOperation batch, CancellationToken cancellationToken)
```

Comment:
Since you're not doing anything special with the `CreateIfNotExistsAsync` call here (and it looks like it's only called from one place) -- can you remove this method and instead wrap the call to `ExecuteBatchAsync` in-line?